src/core/components/modal/index.js

```diff
@@ -0,0 +1,152 @@
+// @flow
+
+/**
+ * Copyright (c) Garuda Labs, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { Animated, Modal, StyleSheet, View } from 'react-native';
+import React, { useRef, useState } from 'react';
+import type { LayoutEvent } from 'react-native/Libraries/Types/CoreEventTypes';
+import ModalButton from './modal-button';
+import type { Node } from 'react';
+import Overlay from './overlay';
+import type { Props } from './types';
+import type { StyleSheet as StyleSheetType } from 'hyperview/src/types';
+import { createStyleProp } from 'hyperview/src/services';
+import styles from './styles';
+
+/**
+ * Renders a bottom sheet with cancel/done buttons and a picker component.
+ * Uses styles defined on the <picker-field> element for the modal and buttons.
+ * This is used on iOS only.
+ */
+export default (props: Props): Node => {
+  const [visible, setVisible] = useState(props.isFocused());
+  const [height, setHeight] = useState(0);
+
+  const translateY = useRef(new Animated.Value(0)).current;
+  const opacity = useRef(new Animated.Value(0)).current;
+
+  const style: Array<StyleSheetType> = createStyleProp(
+    props.element,
+    props.stylesheets,
+    {
+      ...props.options,
+      styleAttr: 'modal-style',
+    },
+  );
+
+  const cancelLabel: string =
+    props.element.getAttribute('cancel-label') || 'Cancel';
+  const doneLabel: string = props.element.getAttribute('done-label') || 'Done';
+
+  const getTextStyle = (pressed: boolean): Array<StyleSheetType> =>
+    createStyleProp(props.element, props.stylesheets, {
+      ...props.options,
+      pressed,
+      styleAttr: 'modal-text-style',
+    });
+
+  const overlayStyle = StyleSheet.flatten(
+    createStyleProp(props.element, props.stylesheets, {
+      ...props.options,
+      styleAttr: 'modal-overlay-style',
+    }),
+  );
+
+  const onLayout = (event: LayoutEvent) => {
+    setHeight(event.nativeEvent.layout.height);
+  };
+
+  const animationDuration: number =
+    parseInt(props.element.getAttribute('modal-animation-duration'), 10) || 250;
```

Comment:
this might be surprising behavior if someone sets the duration to `0` (they want no animation) and it defaults to 250.

Reply:
good catch, I'll fix this

Reply:
addressed in #655